chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -0,0 +1,25 @@
+// See LICENSE for license details.
+
+package chisel3.core
+
+import scala.language.experimental.macros
+
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.{SourceInfo}
+
+private[chisel3] final case class ClockAndReset(clock: Clock, reset: Bool)
+
+object withClockAndReset {  // scalastyle:ignore object.name
+  def apply[T](clock: Clock, reset: Bool)(block: => T)(implicit sourceInfo: SourceInfo): T = {
```

Comment:
@ducky64 I currently have `(implicit sourceInfo: SourceInfo)` at the end but since nothing in the function needs sourceInfo I'm guessing that's unnecessary right? The code in the block should get the sourceInfo on it's own?